src/theory/uf/theory_uf.cpp

```diff
@@ -321,7 +321,7 @@ Node TheoryUF::explain(TNode literal) {
   return explain(literal, NULL);
 }
 
-Node TheoryUF::explain(TNode literal, eq::EqProof* pf) {
+Node TheoryUF::explain(TNode literal, std::shared_ptr<eq::EqProof> pf) {
```

Comment:
It seems like `std::shared_ptr<EqProof> eqp` should be a `std::weak_ptr` since `explain` is not taking (shared) ownership.